clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5522,6 +5522,11 @@ bool Sema::CheckCallingConvAttr(const ParsedAttr &Attrs, CallingConv &CC,
       A = HostTI->checkCallingConvention(CC);
     if (A == TargetInfo::CCCR_OK && CheckDevice && DeviceTI)
       A = DeviceTI->checkCallingConvention(CC);
+  } else if (LangOpts.SYCLIsDevice && TI.getTriple().isAMDGPU() &&
```

Comment:
maybe add a comment: assuming function with CC_X86VectorCall calling convention is always to be emitted on host target.

Reply:
Will add, thanks for the review!